src/recommendpolicy/recommendPolicy.go

```diff
@@ -223,7 +205,55 @@ func uniqueNsDeploy(deployName, deployNamespace string) *types.Deployment {
 	return &deploy
 }
 
-func GetHardenPolicy(deployments *v1.DeploymentList, replicaSets *v1.ReplicaSetList, statefulSets *v1.StatefulSetList, daemonSets *v1.DaemonSetList, nsNotFilter []string) []types.KnoxSystemPolicy {
+func GetAdmissionControllerPolicy(deployments, replicaSets, statefulSets, daemonSets []types.Deployment) []v1.Policy {
+
+	var policies []v1.Policy
+
+	nsNotFilterAdmissionControllerPolicy := cfg.CurrentCfg.ConfigAdmissionControllerPolicy.NsNotFilter
+	nsFilterAdmissionControllerPolicy := cfg.CurrentCfg.ConfigAdmissionControllerPolicy.NsFilter
+	recommendAdmissionControllerPolicy := cfg.GetCfgRecommendAdmissionControllerPolicy()
+
+	for _, d := range deployments {
+
+		labelMap := labelArrayToLabelMap(strings.Split(d.Labels, ","))
+
+		if recommendAdmissionControllerPolicy &&
+			isNamespaceAllowed(d.Namespace, nsNotFilterAdmissionControllerPolicy, nsFilterAdmissionControllerPolicy) {
+			policies = append(policies, generateAdmissionControllerPolicy(d.Name, d.Namespace, labelMap)...)
+		}
+	}
+	for _, d := range replicaSets {
+
+		labelMap := labelArrayToLabelMap(strings.Split(d.Labels, ","))
+
+		if recommendAdmissionControllerPolicy &&
+			isNamespaceAllowed(d.Namespace, nsNotFilterAdmissionControllerPolicy, nsFilterAdmissionControllerPolicy) {
+			policies = append(policies, generateAdmissionControllerPolicy(d.Name, d.Namespace, labelMap)...)
+		}
+	}
+	for _, d := range statefulSets {
+
+		labelMap := labelArrayToLabelMap(strings.Split(d.Labels, ","))
+
+		if recommendAdmissionControllerPolicy &&
+			isNamespaceAllowed(d.Namespace, nsNotFilterAdmissionControllerPolicy, nsFilterAdmissionControllerPolicy) {
+			policies = append(policies, generateAdmissionControllerPolicy(d.Name, d.Namespace, labelMap)...)
+		}
+	}
+	for _, d := range daemonSets {
+
+		labelMap := labelArrayToLabelMap(strings.Split(d.Labels, ","))
+
+		if recommendAdmissionControllerPolicy &&
+			isNamespaceAllowed(d.Namespace, nsNotFilterAdmissionControllerPolicy, nsFilterAdmissionControllerPolicy) {
+			policies = append(policies, generateAdmissionControllerPolicy(d.Name, d.Namespace, labelMap)...)
+		}
+	}
+
+	return policies
+}
+
+func GetHardenPolicy(deployments, replicaSets, statefulSets, daemonSets []types.Deployment, nsNotFilter []string) []types.KnoxSystemPolicy {
 
 	var policies []types.KnoxSystemPolicy
 	if !isLatest() {
```

Comment:
Not related to the current changes but can you please move the logic of `DownloadAndUnzipRelease` out of this function? It is not just related to hardening policy and results in this function doing more than what its name suggests.